app/CMakeLists.txt

```diff
@@ -37,17 +37,15 @@ if(ANDROID)
     endif()
 
     add_library(${PROJECT_NAME} SHARED ${SRC})
-elseif (APPLE)
-    if(IOS)
-        list(REMOVE_AT SRC 0)
-        list(APPEND SRC
-                ${CMAKE_CURRENT_SOURCE_DIR}/ios/main.mm
-                ${CMAKE_CURRENT_SOURCE_DIR}/ios/AppDelegate.h
-                ${CMAKE_CURRENT_SOURCE_DIR}/ios/AppDelegate.m
-                ${CMAKE_CURRENT_SOURCE_DIR}/ios/ViewController.h
-                ${CMAKE_CURRENT_SOURCE_DIR}/ios/ViewController.mm
-        )
-    endif ()
+elseif(IOS)
```

Comment:
Is this logic change correct?  It looks like it went from adding ios specific files before defining the executable as a macosx bundle when it's an ios file but when it's just apple, it's a macosx bundle with the changes it looks like we're treating osx bundles as a win32.

Reply:
Yes, this is correct.  You only want an app bundle (`MACOSX_BUNDLE`) for iOS since it is required there.  On macOS using an app bundle is awkward since **vulkan_samples** is fundamentally a command line application with command line options.  Actually your current documentation assumes it's a command line app.  When making a command line application on macOS or linux, the cmake option WIN32 is ignored.  Note this cmake code path (without my change) currently supports linux, even though `add_executable()` specifies `WIN32`.

Reply:
I think this is correct for the current architecture.  I think that we should modify the OSX application logic to be useful in a MACOSX_BUNDLE rather than a standard add_executable() call.  That's more purely for the fact that it would be more idomatic.  However, I think upon review, I like your approach better.